lib/src/command/remove.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:yaml_edit/yaml_edit.dart';
+
+import '../command.dart';
+import '../exceptions.dart';
+import '../io.dart';
+
+/// Handles the `remove` pub command.
+class RemoveCommand extends PubCommand {
+  @override
+  String get name => 'remove';
+  @override
+  String get description => 'Removes a dependency from the current package.';
+  @override
+  String get invocation => 'pub remove <package>';
+  @override
+  String get docUrl => 'https://dart.dev/tools/pub/cmd/pub-remove';
+
+  RemoveCommand();
+
+  @override
+  Future run() async {
+    if (argResults.rest.isEmpty) {
+      usageException('Must specify a package to be removed.');
+    }
+
+    /// Update the pubspec.
+    _removePackagesFromPubspec(argResults.rest);
+
+    /// Run pub get once we have successfully updated the pubspec
+    await runner.run(['get']);
+  }
+
+  /// Writes the changes to the pubspec file
+  void _removePackagesFromPubspec(List<String> packages) {
+    ArgumentError.checkNotNull(packages, 'packages');
+
+    if (entrypoint.pubspecPath == null) {
```

Comment:
the `entrypoint` getter will throw if there is not pubspec.yaml. You don't need this check.